src/RA_AchievementSet.cpp

```diff
@@ -431,38 +364,42 @@ BOOL AchievementSet::LoadFromFile(ra::GameID nGameID)
     const std::string sFilename = GetAchievementSetFilename(nGameID);
 
     SetCurrentDirectory(NativeStr(g_sHomeDir).c_str());
-    FILE* pFile = nullptr;
-    errno_t nErr = fopen_s(&pFile, sFilename.c_str(), "r");
-    if (pFile != nullptr)
+
+    if (std::ifstream ifile{ sFilename }; !ifile.is_open())
+    {
+        //	Cannot open file
+        RA_LOG("Cannot open file %s\n", sFilename.c_str());
+        auto errBuf{ std::make_unique<char[]>(1024) };
+        strerror_s(errBuf.get(), 1024, errno);
+        RA_LOG("Error %s\n", errBuf.get());
+        ra::ShowError(errBuf.get());
+        return FALSE;
+    }
+    else
     {
         //	Store this: we are now assuming this is the correct checksum if we have a file for it
         g_pCurrentGameData->SetGameID(nGameID);
 
-        if (m_nSetType == Local)
+        if (m_nSetType == AchievementSetType::Local)
         {
-            const char EndLine = '\n';
-            char buffer[4096];
-            unsigned long nCharsRead = 0;
+            auto buffer{ std::make_unique<char[]>(2048) };
 
             //	Get min ver:
-            _ReadTil(EndLine, buffer, 4096, &nCharsRead, pFile);
+            ifile.getline(buffer.get(), 2048);
```

Comment:
OMG, why!?

I indicate this code will have to change for #82 in response to some comment you had about using make_unique to allocate a buffer and you interpret that as the buffer was too large? Did you even read #82? The problem is that the line was too big to fit in the buffer, so a smaller buffer is not the solution, nor should you be trying to fix it in this PR.

Reply:
I did read the issue and misunderstood. I will consult with you if I'm not sure what you mean.